### PR TITLE
Fix unnecessary checks in the package build

### PR DESCRIPTION
**Pull request by Nuvindu**
## Purpose
Fixes: https://github.com/ballerina-platform/plugin-gradle/issues/128

### CODE REVIEW

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy [hunk 1]
```diff
@@ -276,123 +265,121 @@ class BallerinaPlugin implements Plugin<Project> {
...
-                            createDockerEnvFile("$project.projectDir/docker.env")
-                            def balPackWithDocker = """
+                            def balTestWithDocker = """
                                 docker run --env-file $project.projectDir/docker.env --rm --net=host -u root \
                                     -v $parentDirectory:/home/ballerina/$parentDirectory.name \
                                     -v $projectDirectory:/home/ballerina/$parentDirectory.name/$projectDirectory.name \
                                     ballerina/ballerina:$ballerinaDockerTag \
                                     /bin/sh -c "cd $parentDirectory.name/$projectDirectory.name && \
-                                    $balJavaDebugParam bal pack --target-dir ${balBuildTarget} ${debugParams}"
+                                    bal test ${graalvmFlag} ${testCoverageParams} ${groupParams} ${disableGroups} ${debugParams}"
```
Comment (ThisaruGuruge) [inline src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy:307]: We shouldn't remove the `$balJavaDebugParam` here.

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy [hunk 2]
```diff
@@ -276,123 +265,121 @@ class BallerinaPlugin implements Plugin<Project> {
...
+                        if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                            commandLine 'cmd', '/c', "$balPackWithDocker"
+                        } else {
+                            commandLine 'sh', '-c', "$balPackWithDocker"
+                        }
+                    } else {
+                        if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                            commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat pack --target-dir ${balBuildTarget} --offline ${debugParams} && exit %%ERRORLEVEL%%"
+                        } else {
+                            commandLine 'sh', '-c', "$balJavaDebugParam $distributionBinPath/bal pack --target-dir ${balBuildTarget} --offline ${debugParams}"
```
Comment (ThisaruGuruge) [inline src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy:291]: For `bal pack` command, we can remove the `$balJavaDebugParam` and `${debugParams}`.

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy [hunk 3]
```diff
@@ -276,123 +265,121 @@ class BallerinaPlugin implements Plugin<Project> {
...
+                    environment 'JAVA_OPTS', '-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true'
+                    if (buildOnDocker) {
+                        createDockerEnvFile("$project.projectDir/docker.env")
+                        def balPackWithDocker = """
+                            docker run --env-file $project.projectDir/docker.env --rm --net=host -u root \
+                                -v $parentDirectory:/home/ballerina/$parentDirectory.name \
+                                -v $projectDirectory:/home/ballerina/$parentDirectory.name/$projectDirectory.name \
+                                ballerina/ballerina:$ballerinaDockerTag \
+                                /bin/sh -c "cd $parentDirectory.name/$projectDirectory.name && \
+                                $balJavaDebugParam bal pack --target-dir ${balBuildTarget} ${debugParams}"
```
Comment (ThisaruGuruge) [inline src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy:280]: We don't need the `$balJavaDebugParam` here. It should be used in `test` task.

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy (hunk 1 above)
Comment (Nuvindu) [reply src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy]: Fixed in 2e77861

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy (hunk 2 above)
Comment (Nuvindu) [reply src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy]: Fixed in 2e77861

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy (hunk 3 above)
Comment (Nuvindu) [reply src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy]: Fixed in 2e77861

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy [hunk 4]
```diff
@@ -276,123 +265,121 @@ class BallerinaPlugin implements Plugin<Project> {
...
+                    } else {
+                        if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                            commandLine 'cmd', '/c', "$distributionBinPath/bal.bat pack --target-dir ${balBuildTarget} --offline && exit %%ERRORLEVEL%%"
+                        } else {
+                            commandLine 'sh', '-c', "$distributionBinPath/bal pack --target-dir ${balBuildTarget} --offline"
+                        }
+                    }
+                }
+                // Run tests
+                if (!skipTests) {
```
Comment (ThisaruGuruge) [inline src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy:296]: We should run the tests inside the `test` task. Now the `test` task is empty and the tests are being run inside the `build` task. Shall we move this to the `test` task?

Comment (Nuvindu) [reply src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy]: Fixed in [af88463](https://github.com/ballerina-platform/plugin-gradle/pull/129/commits/af88463c788767d6497fbe5ee8f12eaeaf7ed197)

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**